imports/client/components/FirehosePage.tsx

```diff
@@ -0,0 +1,267 @@
+    const searchKeys = getSearchString().split(' ');
+    let interestingChatMessages;
+
+    if (searchKeys.length === 1 && searchKeys[0] === '') {
+      interestingChatMessages = chatMessages;
+    } else {
+      const searchKeysWithEmptyKeysRemoved = searchKeys.filter((key) => { return key.length > 0; });
+      const isInteresting = compileMatcher(searchKeysWithEmptyKeysRemoved);
+      interestingChatMessages = chatMessages.filter(isInteresting);
+    }
+
+    return interestingChatMessages;
+  }, [getSearchString, compileMatcher]);
+
+  const [shouldScrollBottom, setShouldScrollBottom] = useState<boolean>(true);
+
+  const saveShouldScroll = useCallback(() => {
+    const pane = messagesPaneRef.current;
+    if (pane) {
+      const basicallyAtBottom = pane.clientHeight + pane.scrollTop + 5 >= pane.scrollHeight;
+      setShouldScrollBottom(basicallyAtBottom);
+    }
+  }, []);
+
+  const forceScrollBottom = useCallback(() => {
+    const pane = messagesPaneRef.current;
+    if (pane) {
+      pane.scrollTop = pane.scrollHeight;
+    }
+  }, []);
+  const maybeForceScrollBottom = useCallback(() => {
+    if (shouldScrollBottom) {
+      forceScrollBottom();
+    }
+  }, [shouldScrollBottom, forceScrollBottom]);
+
+  const chats = useMemo(() => {
+    return filteredChats(chatMessagesTracker.chatMessages);
+  }, [filteredChats, chatMessagesTracker.chatMessages]);
+
+  const ready = profilesTracker.ready && puzzlesTracker.ready && chatMessagesTracker.ready;
+
+  useEffect(() => {
+    maybeForceScrollBottom();
+  }, [ready, maybeForceScrollBottom]);
+
+  if (!ready) {
+    return <div>loading...</div>;
+  }
+
+  return (
+    <FixedLayout>
+      <FirehosePageLayout>
+        <h1>Firehose (all chat messages, expect slowness)</h1>
+        <FormGroup>
+          <InputGroup>
+            <FormControl
+              id="jr-firehose-search"
+              as="input"
+              type="text"
+              ref={searchBarRef}
+              placeholder="Filter by message contents"
+              value={getSearchString()}
+              onChange={onSearchStringChange}
+            />
+            <InputGroup.Append>
+              <Button variant="secondary" onClick={clearSearch}>
+                <FontAwesomeIcon icon={faEraser} />
+              </Button>
+            </InputGroup.Append>
+          </InputGroup>
+        </FormGroup>
+        <MessagesPane ref={messagesPaneRef} onScroll={saveShouldScroll} className={shouldScrollBottom ? 'live' : ''}>
+          {chats.map((msg) => {
+            return (
+              <Message
+                key={msg._id}
+                msg={msg}
+                puzzles={puzzlesTracker.puzzles}
+                displayNames={profilesTracker.displayNames}
+              />
+            );
+          })}
+        </MessagesPane>
+        <div>
+          {shouldScrollBottom ? 'Log will update to show new messages as they arrive' : ''}
```

Comment:
I found it took me a while to understand what this presentation was conveying. In fact, I initially thought the disappearing text was a glitch. I would prefer a persistent display of state, such as "Live scrolling: [On/Off]".

Frankly, since the behavior is pretty intuitive and consistent with puzzle page chat histories, I'm not sure I see the need for this message at all.

Reply:
Sure, that seems reasonable.  I originally added it with the intent of making it a little more visible if you weren't actually following the tail of the log (and might have been expecting it to) and then missed something, but I suppose if we do a good enough job with the behavior it's not needed.